pkg/controllers/networkinfo/networkinfo_controller.go

```diff
@@ -36,6 +35,54 @@ var (
 	MetricResType = common.MetricResTypeNetworkInfo
 )
 
+const (
+	NamespaceNetworkReady corev1.NamespaceConditionType = "NamespaceNetworkReady"
+
+	NSReasonVPCNetConfigNotReady string = "VPCNetworkConfigurationNotReady"
+	NSReasonVPCNotReady          string = "VPCNotReady"
+	NSReasonVPCSnatNotReady      string = "VPCSnatNotReady"
+)
+
+var (
+	nsMsgVPCNetCfgGetError        = newNsUnReadyMessage("Error happened to get VPC network configuration: %v", NSReasonVPCNetConfigNotReady)
+	nsMsgSystemVPCNetCfgNotFound  = newNsUnReadyMessage("Error happened to get system VPC network configuration: %v", NSReasonVPCNetConfigNotReady)
+	nsMsgVPCGwConnectionGetError  = newNsUnReadyMessage("Error happened to validate system VPC gateway connection readiness: %v", NSReasonVPCNetConfigNotReady)
+	nsMsgVPCGwConnectionNotReady  = newNsUnReadyMessage("System VPC gateway connection is not ready", NSReasonVPCNetConfigNotReady)
+	nsMsgVPCCreateUpdateError     = newNsUnReadyMessage("Error happened to create or update VPC: %v", NSReasonVPCNotReady)
+	nsMsgVPCNsxLBSNotReady        = newNsUnReadyMessage("Error happened to get NSX LBS path in VPC: %v", NSReasonVPCNotReady)
+	nsMsgVPCAviSubnetError        = newNsUnReadyMessage("Error happened to get Avi Load balancer Subnet info: %v", NSReasonVPCNotReady)
+	nsMsgVPCGetExtIPBlockError    = newNsUnReadyMessage("Error happened to get external IP blocks: %v", NSReasonVPCNotReady)
+	nsMsgVPCNoExternalIPBlock     = newNsUnReadyMessage("System VPC has no external IP blocks", NSReasonVPCNotReady)
+	nsMsgVPCAutoSNATDisabled      = newNsUnReadyMessage("SNAT is not enabled in System VPC", NSReasonVPCSnatNotReady)
+	nsMsgVPCDefaultSNATIPGetError = newNsUnReadyMessage("Default SNAT IP is not allocated in VPC: %v", NSReasonVPCSnatNotReady)
+	nsMsgVPCIsReady               = newNsUnReadyMessage("", "")
+)
+
+type nsUnReadyMessage struct {
+	reason string
+	msg    string
+}
+
+func newNsUnReadyMessage(msg string, reason string) *nsUnReadyMessage {
+	return &nsUnReadyMessage{
+		msg:    msg,
+		reason: reason,
+	}
+}
+
+func (m *nsUnReadyMessage) getNSNetworkCondition(options ...interface{}) *corev1.NamespaceCondition {
+	cond := &corev1.NamespaceCondition{
+		Type:   NamespaceNetworkReady,
+		Status: corev1.ConditionTrue,
+	}
+	if m.reason != "" {
+		cond.Status = corev1.ConditionFalse
+		cond.Reason = m.reason
```

Comment:
if status is false, the Type is still  NamespaceNetworkReady?

Reply:
Yes, the Namespace condition type is "NamespaceNetworkReady", we use field "status" to express the Condition is ready or not.